qa-tests-backend/src/main/groovy/services/RoleService.groovy

```diff
@@ -55,14 +69,18 @@ class RoleService extends BaseService {
                 .setRole(role)
                 .build()
         )
+        log.info "Created role: ${role.name}"
         role
     }
 
-    static deleteRole(String name) {
+    static deleteRole(String name, Boolean alsoDeletePermissionSet = true) {
         try {
             def role = getRole(name)
             getRoleService().deleteRole(Common.ResourceByID.newBuilder().setId(name).build())
-            deletePermissionSet(role.permissionSetId)
+            if (alsoDeletePermissionSet) {
+                deletePermissionSet(role.permissionSetId)
+            }
+            log.info "Deleted role: ${name}, and permission set (${alsoDeletePermissionSet})"
```

Comment:
I think something like the following would be easier to understand than `Deleted role foo and permission set (false)`:
```suggestion
            log.info "Deleted role: ${name}" + (alsoDeletePermissionSet ? ", and permission set" : "")
```